### PR TITLE
[full-ci] StatusIndicators in ResourceTable Followup

### PR DESCRIPTION
**Pull request by lookacat**
## Description
See https://github.com/owncloud/web/issues/5976

## Related Issue

- Fixes https://github.com/owncloud/web/issues/5976

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/23820/10/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23820/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2022-3-17-12-56-08.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23820/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-brian-2022-3-17-12-56-30.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23820/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-carol-2022-3-17-12-56-37.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23820/tracing/file-alice-2022-3-17-12-54-20.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23820/tracing/file-brian-2022-3-17-12-54-29.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23820/tracing/folder-alice-2022-3-17-12-52-46.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23820/tracing/folder-brian-2022-3-17-12-52-55.zip 


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/23843/10/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23843/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2022-3-17-07-09-54.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23843/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-brian-2022-3-17-07-10-17.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23843/tracing/file-alice-2022-3-17-07-08-07.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23843/tracing/file-brian-2022-3-17-07-08-16.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23843/tracing/folder-alice-2022-3-17-07-06-35.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23843/tracing/folder-brian-2022-3-17-07-06-43.zip 


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/23846/10/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23846/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2022-3-17-07-39-57.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23846/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-brian-2022-3-17-07-40-19.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23846/tracing/file-alice-2022-3-17-07-38-05.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23846/tracing/file-brian-2022-3-17-07-38-13.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23846/tracing/folder-alice-2022-3-17-07-36-33.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23846/tracing/folder-brian-2022-3-17-07-36-41.zip 


</p></details>

File: changelog/unreleased/enhancement-move-share-indicators [hunk 1]
```diff
@@ -1,9 +1,11 @@
 Enhancement: Move share indicators
 
 We've moved the share/status indicators into a separate column
-and adjusted the design in ODS.
+and adjusted the design in ODS. We also removed links from
+shared-with-me-page.
```
Comment (kulmann) [inline changelog/unreleased/enhancement-move-share-indicators:5]: Shared with others - but please create a separate changelog items for this enhancement (the removal of links from shared with others).

**Review by kulmann — CHANGES_REQUESTED** [review]
Please rebase to make e2e tests pass again. Also see comment for regarding changelog, I think the removal of links from shared with others deserves its own changelog item because it has nothing to do with the separate column for share indicators.

File: packages/web-app-files/src/components/FilesList/ResourceTable.vue [hunk 2]
```diff
@@ -421,7 +426,15 @@ export default {
...
         ]
-          .filter((field) => Object.prototype.hasOwnProperty.call(firstResource, field.name))
+          .filter((field) => {
+            if (!Object.prototype.hasOwnProperty.call(firstResource, field.name)) {
+              return false
+            }
+            if (field.name !== 'indicators') {
+              return true
+            }
+            return firstResource.indicators.length
```
Comment (kulmann) [inline packages/web-app-files/src/components/FilesList/ResourceTable.vue:436]: Why only on the first resource?!

File: packages/web-app-files/src/components/FilesList/ResourceTable.vue [hunk 3]
```diff
@@ -472,6 +485,14 @@ export default {
     }
   },
   methods: {
+    ...mapActions('Files/sidebar', ['openWithPanel']),
+    openSharingSidebar(file) {
+      if (file.shareTypes?.includes(ShareTypes.link.value)) {
```
Comment (kulmann) [inline packages/web-app-files/src/components/FilesList/ResourceTable.vue:490]: I added this myself, but the shareTypes field is not available when the function is called. Need to check on monday.

**Review by kulmann — CHANGES_REQUESTED** [review]

File: packages/web-app-files/src/components/FilesList/ResourceTable.vue (hunk 3 above)
Comment (lookacat) [reply packages/web-app-files/src/components/FilesList/ResourceTable.vue:490]: seems to work fine

Comment (kulmann) [reply packages/web-app-files/src/components/FilesList/ResourceTable.vue:490]: for me the `shareTypes` field is not available in the moment the button is clicked

Comment (lookacat) [reply packages/web-app-files/src/components/FilesList/ResourceTable.vue]: @kulmann fixed it by using share?.shareType

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/23935/10/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23935/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2022-3-21-01-57-31.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23935/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-brian-2022-3-21-01-58-00.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23935/tracing/file-alice-2022-3-21-01-55-10.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23935/tracing/file-brian-2022-3-21-01-55-22.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23935/tracing/folder-alice-2022-3-21-01-53-24.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23935/tracing/folder-brian-2022-3-21-01-53-35.zip 


</p></details>

**Review by kulmann — APPROVED** [review]
Needs a rebase, but looks good otherwise 👍

**Comment by pascalwengerter** [comment]
> Running the latest CI failure (https://drone.owncloud.com/owncloud/web/24011/44/20) locally as `SCREEN_RESOLUTION=768x1024 yarn test:acceptance:oc10 features/webUIRestrictSharing/restrictReSharing.feature:23` (pasting it here in case I miss something obvious) fails with "resource can't be favorited in the server" - which is not what the CI fails with. @kulmann @fschade any ideas?

Tried dropping volumes & pulling newer images, problem stays

**Comment by kulmann** [comment]
> > Running the latest CI failure (https://drone.owncloud.com/owncloud/web/24011/44/20) locally as `SCREEN_RESOLUTION=768x1024 yarn test:acceptance:oc10 features/webUIRestrictSharing/restrictReSharing.feature:23` (pasting it here in case I miss something obvious) fails with "resource can't be favorited in the server" - which is not what the CI fails with. @kulmann @fschade any ideas?
> 
> Tried dropping volumes & pulling newer images, problem stays

might be a timing issue, that the share is not yet accepted. try moving the line that sets the favorite some lines down (below the other middleware step).

**Review by pascalwengerter — APPROVED** [review]
Rebased, and highly in favor of merging this one if CI gets green finally. There's some usability things to think of (e.g. clicking on share giver in the shared-with-me view leads to the sharing tab on the sidebar that simply reads "No permissions to share file/folder", but we can fix that at a later stage)

**Pull request merged**